src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -89,6 +89,36 @@ def set_top_tags(
         logger.exception("We failed to set a tag.")
 
 
+def update_result_and_tags(
```

Comment:
Everything in this function is mostly what was outside of the for loop in the main function.